test/server/guarddog_impl_test.cc

```diff
@@ -174,6 +206,38 @@ TEST_P(GuardDogAlmostDeadTest, MultiKillNoFinalCheckTest) {
   SetupForMultiDeath();
 }
 
+TEST_P(GuardDogDeathTest, MultiKillThresholdDeathTest) {
+  auto die_function = [&]() -> void {
+    setupForMultiDeathThreshold();
+
+    // Pet the last two dogs so we're just at the threshold that causes death.
+    dogs_.at(4)->touch();
+    dogs_.at(3)->touch();
+
+    time_system_->advanceTimeWait(std::chrono::milliseconds(2)); // 1 ms past multi-death.
+    guard_dog_->forceCheckForTest();
+  };
+  EXPECT_DEATH(die_function(), "");
+}
+
+TEST_P(GuardDogAlmostDeadTest, MultiKillUnderThreshold) {
+  // This does everything the death test does except it pets an additional watchdog
+  // that causes us to be under the threshold (60%) of multikill death.
+  auto die_function = [&]() -> void {
+    setupForMultiDeathThreshold();
+
+    // Pet the last three dogs so we're just under the threshold that causes death.
+    dogs_.at(4)->touch();
+    dogs_.at(3)->touch();
+    dogs_.at(2)->touch();
+
+    time_system_->advanceTimeWait(std::chrono::milliseconds(2)); // 1 ms past multi-death.
+    guard_dog_->forceCheckForTest();
+  };
+
+  die_function();
```

Comment:
nit: you don't need to wrap in die_function in this case, since you're not using EXPECT_DEATH